apps/src/gamelab/animationListModule.js

```diff
@@ -687,14 +688,35 @@ function loadPendingFramesFromSource(key, props, callback) {
   };
 }
 
+/**
+  * Given the animationList and key, determine whether the animation identified
+  * by the key currently exists in the project. If not present, the animation
+  * could be deleted or not a part of this project.
+*/
+function sourcePresentInProject(animationList, key) {
+  let inProject = false;
+  const props = animationList.propsByKey[key];
+  animationList.orderedKeys.forEach(item=> {
+      if (props.sourceUrl.includes(item)) {
+        inProject = true;
+      }
+  });
+  return inProject;
+}
+
 /**
  * Given a key/serialized-props pair for an animation, work out where to get
  * the spritesheet.
  * @param {!AnimationKey} key
  * @param {!SerializedAnimationProps} props
+ * @param {boolean} withVersion - Whether to request a specific version of the
+ *        animation if pulling from the local project.
+ * @param {string} channelId - Used to differentiate library animations from others
+ * @param {object} animationList - Used to check if src url exists in this
+ *        project and is deleted
  * @returns {string}
  */
-export function animationSourceUrl(key, props) {
+export function animationSourceUrl(key, props, withVersion = false, channelId = null, animationList = null) {
   // TODO: (Brad) We want to get to where the client doesn't know much about
   //       animation versions, by switching to Chris' new Files API.
   //       in the meantime, be able to request versions only when we export
```

Comment:
@islemaster Do we need this ToDo anymore? Seems like we're circumventing this with the data-loss work we're doing.

Reply:
No, go ahead and tear it out.  The first half is still true but not something we're likely to address in the near future.  The "in the meantime" bit doesn't apply anymore.